infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -404,20 +379,21 @@ public void clearLocalStreams() {
 
         log.debug("Total of {} streams were replicated from active out of {}, sequencer query for {}, streamsToQuery={}",
                 replicatedStreamIds.size(), streamViewMap.size(), streamsToQuery.size(), streamsToQuery);
-        Set<UUID> streamsToClear = new HashSet<>();
         TokenResponse tokenResponse = rt.getSequencerView().query(streamsToQuery.toArray(new UUID[0]));
+        Set<UUID> streamsWithLocalWrites = new HashSet<>();
         streamsToQuery.forEach(streamId -> {
             if (tokenResponse.getStreamTail(streamId) != Address.NON_EXIST) {
-                streamsToClear.add(streamId);
+                streamsWithLocalWrites.add(streamId);
             }
         });
 
-        if (!streamsToClear.isEmpty()) {
-            log.debug("Clear streams with local writes, total={}, streams={}", streamsToClear.size(), streamsToClear);
-            clearStreams(streamsToClear);
+        if (!streamsWithLocalWrites.isEmpty()) {
+            log.debug("Clear streams with local writes, total={}, streams={}", streamsWithLocalWrites.size(), streamsWithLocalWrites);
+            streamsToClear.addAll(streamsWithLocalWrites);
```

Comment:
created streamsWithLocalWrites to retain the logging of streams with local writes that may help in debugging.